cmake/Modules/FindGflags.cmake

```diff
@@ -0,0 +1,66 @@
+###############################################################################
+#
+# CMake script for finding the GFlags library.
+#
+# https://gflags.github.io/gflags/
+#
+# Redistribution and use is allowed according to the terms of the 3-clause BSD
+# license.
+#
+# Input variables:
+#
+#
+# Cache variables (not intended to be used in CMakeLists.txt files)
+#
+# - _gflags_ORIG_CMAKE_FIND_LIBRARY_SUFFIXES
+# - CMAKE_FIND_LIBRARY_SUFFIXES: ?
+#
+# Output variables:
+#
+# - Gflags_FOUND: Boolean that indicates if the package was found
+# - Gflags_LIBRARIES: shared objects exported from gflags
+# - Gflags_INCLUDE_DIRS: this is exported from somewhere else magically
+#
+# Example usage:
+#
+#   #Gflags
+#   find_package(Gflags REQUIRED)
+#
+#   include_directories(
+#     ${Gflags_INCLUDE_DIRS}
+#   )
+#
+###############################################################################
+
+
+set(_gflags_ORIG_CMAKE_FIND_LIBRARY_SUFFIXES ${CMAKE_FIND_LIBRARY_SUFFIXES})
+
+find_path(Gflags_INCLUDE_PATH gflags/gflags.h)
+
+if (Gflags_STATIC)
+  if (WIN32)
+    set(CMAKE_FIND_LIBRARY_SUFFIXES .lib ${CMAKE_FIND_LIBRARY_SUFFIXES})
+  else (WIN32)
+    set(CMAKE_FIND_LIBRARY_SUFFIXES .a ${CMAKE_FIND_LIBRARY_SUFFIXES})
+  endif (WIN32)
+endif (Gflags_STATIC)
+
+find_library(Gflags_LIBRARY NAMES gflags libgflags)
+
+if(Gflags_INCLUDE_PATH AND Gflags_LIBRARY)
+  set(Gflags_FOUND TRUE)
+endif(Gflags_INCLUDE_PATH AND Gflags_LIBRARY)
+
+if(Gflags_FOUND)
+  if(NOT Gflags_FIND_QUIETLY)
+    message(STATUS "Found gflags: ${Gflags_LIBRARY}")
+  endif(NOT Gflags_FIND_QUIETLY)
+else(Gflags_FOUND)
+  if(Gflags_FIND_REQUIRED)
+    message(FATAL_ERROR "Could not find gflags library.")
+  endif(Gflags_FIND_REQUIRED)
+endif(Gflags_FOUND)
+
+set(CMAKE_FIND_LIBRARY_SUFFIXES ${_gflags_ORIG_CMAKE_FIND_LIBRARY_SUFFIXES})
+
+set(Gflags_LIBRARIES gflags)
```

Comment:
The CMake code above goes to the trouble of finding the actual `gflags` library with a call to `find_library()` and then you just set `Gflags_LIBRARIES` to `gflags` blindly. That doesn't seem right.